build/templates/session.py.mako

```diff
@@ -143,6 +145,13 @@ class Session(object):
         self.library = library.get_library()
         self.vi = 0  # This must be set before calling _init_with_options.
         self.vi = self._init_with_options(resource_name, id_query, reset_device, options_string)
+
+        self.__isfrozen = True
+
+    def __setattr__(self, key, value):
+        if self.__isfrozen and key not in dir(self):
+            raise TypeError("%r is a frozen class" % self)
```

Comment:
Should be KeyError or AttributeError (not sure which makes more sense semantically, would need to check docs)